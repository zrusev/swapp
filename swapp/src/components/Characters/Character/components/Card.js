import styled from 'styled-components';

export const Card = styled.div`
  flex: 3;
  display: flex;
  flex-direction: column;
  align-items: center;
  background: ${({ theme: { cardBackground } }) => cardBackground};
  border-color: ${({ theme: { cardBorderColor } }) => cardBorderColor};
  text-decoration: none;
  box-shadow: rgba(0, 0, 0, 0.19) 0 0 8px 0;
  border-radius: 5px;
  margin: 1em;
`;
