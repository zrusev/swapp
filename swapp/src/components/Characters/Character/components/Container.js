import styled from 'styled-components';

export const Container = styled.div`
    max-width: 65vw;
    margin: 0 auto;
    margin-bottom: 5em;
    display: flex;
    flex-flow: row wrap;
    border-top: 5px solid ${({theme: {lineColor}}) => lineColor};
`;