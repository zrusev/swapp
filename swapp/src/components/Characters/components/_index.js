export * from './Container';
export * from './Person';
export * from './Media';
export * from './People';
