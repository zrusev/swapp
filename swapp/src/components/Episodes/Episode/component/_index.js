export * from './Container';
export * from './Header';
export * from './Title';
export * from './Summary';
export * from './Label';
export * from './PeopleWrapper';
export * from './People';
export * from './Person';
export * from './PersonTitle';
export * from './Media';
