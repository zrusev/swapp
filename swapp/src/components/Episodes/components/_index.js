export * from './Container';
export * from './Wrapper';
export * from './Card';
export * from './Title';
export * from './Summary';
export * from './Meta';
export * from './Media';
export * from './MoreItems';