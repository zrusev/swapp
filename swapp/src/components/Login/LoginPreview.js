import React from 'react';

import { 
    Wrapper, 
    Container, 
    Title, 
    FormWrapper,
    Form, 
    WarningBar, 
    Input, 
    Button
} from './components/_index';

const LoginPreview = () => {  
    return(
        <Wrapper>
            <Container>
                <Title>
                    <h1>SWAPP</h1>
                </Title>
                <FormWrapper>
                    <Form>
                        <WarningBar>Invalid Credentials!</WarningBar>
                        <Input 
                            inputType='text' 
                            placeholder="email"
                        />
                        <Input 
                            inputType='password' 
                            placeholder="password"
                        />
                        <Button 
                            type="submit"
                        >
                            <b>Login</b>
                        </Button>
                    </Form>
                </FormWrapper>
            </Container>
        </Wrapper>
    );
}

export default LoginPreview;