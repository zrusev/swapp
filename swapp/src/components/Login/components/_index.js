export * from './Wrapper';
export * from './Container';
export * from './Title';
export * from './FormWrapper';
export * from './Form';
export * from './WarningBar';
export * from './Input';
export * from './Button';