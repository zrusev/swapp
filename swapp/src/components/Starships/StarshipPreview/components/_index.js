export * from './Wrapper';
export * from './Radar';
