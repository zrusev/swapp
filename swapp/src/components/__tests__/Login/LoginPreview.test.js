import React from 'react';
import { shallow } from 'enzyme';

import { Input } from '../../Login/components/_index';

import LoginPreview from '../../Login/LoginPreview';

describe('login form component', () => {
  const props = {
    login: jest.fn(),
  };
  const wrapper = shallow(<LoginPreview {...props} />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have input fields', () => {
    expect(wrapper.find(Input).length).toEqual(2);
  });

  it('should change email value when typed', () => {
    const input = wrapper.find(Input).at(0);

    input.simulate('change', {
      persist: jest.fn(),
      target: {
        name: 'email',
        value: 'demo@st6.io',
      },
    });

    expect(
      wrapper
        .find(Input)
        .at(0)
        .prop('value'),
    ).toEqual('demo@st6.io');
  });

  it('should change password value when typed', () => {
    const input = wrapper.find(Input).at(1);

    input.simulate('change', {
      persist: jest.fn(),
      target: {
        name: 'password',
        value: 'password',
      },
    });

    expect(
      wrapper
        .find(Input)
        .at(1)
        .prop('value'),
    ).toEqual('password');
  });
});
