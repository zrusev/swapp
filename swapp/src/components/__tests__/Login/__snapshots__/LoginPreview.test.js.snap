// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login form component should match snapshot 1`] = `
<styled.main>
  <styled.div>
    <styled.div>
      <h1>
        SWAPP
      </h1>
    </styled.div>
    <styled.div>
      <styled.form
        onSubmit={[Function]}
      >
        <Styled(styled.input)
          inputType="text"
          name="email"
          onChange={[Function]}
          placeholder="email"
          required={true}
          value=""
        />
        <Styled(styled.input)
          inputType="password"
          name="password"
          onChange={[Function]}
          placeholder="password"
          required={true}
          value=""
        />
        <Styled(styled.button)
          type="submit"
        >
          <b>
            Login
          </b>
        </Styled(styled.button)>
      </styled.form>
    </styled.div>
  </styled.div>
</styled.main>
`;
